### ISSUE TITLE
EXCLUDE argument parsed differently in different linters

### ISSUE THREAD

**Issue by ayrton04** [ament/ament_lint#477]
I think what I am reporting may be related to [this](https://github.com/ament/ament_lint/issues/354) issue, but I'm not positive.

When using `EXCLUDE` for various linters, the argument in question is parsed differently. In `cppcheck`, it is parsed as a [multi_value_keyword](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html):

https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cppcheck/cmake/ament_cppcheck.cmake#L38

This allows the user to provide a _list_ of files or directories to exclude. The same is done for [copyright](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_copyright/cmake/ament_copyright.cmake#L30), [uncrustify](https://github.com/ament/ament_lint/blob/rolling/ament_cmake_uncrustify/cmake/ament_uncrustify.cmake#L42), and possibly others.

All of those behave correctly.

However, some of the linter `CMake` functions have `EXCLUDE` listed as a [one_value_keyword](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html), despite the [documentation](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L29-L30) for those methods suggesting that they accept lists of files and directories. Examples are [cpplint](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L37) and [flake8](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_flake8/cmake/ament_flake8.cmake#L42):

https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L37

These don't function as they should, because the `EXCLUDE` argument is only using _one_ value from the provided list. The reason they sometimes _appear_ to work is because of how the arguments are being parsed. Consider this example:

```
  list(APPEND LIST_OF_FILES path/to/file_a.cpp)
  list(APPEND LIST_OF_FILES path/to/file_b.cpp)
  list(APPEND LIST_OF_FILES path/to/file_c.cpp)

  ament_cpplint(
    EXCLUDE "${LIST_OF_FILES}")
```

This will get parsed as such:

1. `path/to/file_a.cpp` will get [parsed](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L50) as the `EXCLUDE` argument (singular)
2. `path/to/file_b.cpp` and `path/to_file_c.cpp` will be [parsed](https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L62) as the `ARG_UNPARSED_ARGUMENTS`.

The net result in this example is that the final command "works": we still construct a python script argument of `--exclude path/to/file_a.cpp path/to/file_b.cpp path/to/file_c.cpp`, but it only works because we are just tacking on the unparsed arguments to the command.

Now consider this example:

```
  list(APPEND LIST_OF_FILES path/to/file_a.cpp)
  list(APPEND LIST_OF_FILES path/to/file_b.cpp)
  list(APPEND LIST_OF_FILES path/to/file_c.cpp)

  ament_cpplint(
    EXCLUDE "${LIST_OF_FILES}"
    MAX_LINE_LENGTH 120)
```

This will **not** behave as intended. The final command will have arguments `--exclude path/to/file_a.cpp --linelength 120 path/to/file_b.cpp path/to/file_c.cpp`. This will cause the linter to _not_ exclude the last two files.

If we instead make the argument list for the erroneous methods use multi-value keywords (shown here for `cpplint`), it works as intended:

```
  cmake_parse_arguments(ARG "" "MAX_LINE_LENGTH;ROOT;TESTNAME;TIMEOUT" "EXCLUDE;FILTERS" ${ARGN})
```

TL; DR, I think all of the CMake functions for linting should have the `EXCLUDE` parameter as one of the multi-value keywords. I am happy to submit a PR, if that would help, or if a minimal example program is needed, I can provide it. But the difference in parameter parsing seems to be clear from the code.

**Comment by wjwwood** [comment]
From our issue triaging meeting, looks like a bug/oversight which would be nice to fix (make consistent) with multi-value keyword arguments, help implementing it would be appreciated.

**Comment by ayrton04** [comment]
Happy to!

**Comment by sloretz** [comment]
Closing as complete with #481. Thank you for the PR!
